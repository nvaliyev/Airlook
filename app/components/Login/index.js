import React, {useState} from 'react';
import {View, TextInput} from 'react-native';
import styles from './style';
import MButton from '../mButton';
import ROUTES from '../../servises/routes';
import config from '../../servises/utils';

const Login = ({navigation}) => {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const User = {username: 'a', password: 'a'};

  const CheckLogin = () => {
    navigation.navigate(ROUTES.FlightScreen);
  };

  // const CheckLogin = (navigation) => {

  //   if (User.username === username && User.password === password) {
  //     console.log('navigation', navigation)
  //     navigation.navigate(ROUTES.FlightScreen)
  //   }
  //   else {
  //     alert('Username or Password is incorrect')
  //   }
  // }

  return (
    <View style={styles.container}>
      <TextInput
        style={styles.input}
        placeholder="Username:"
        onChangeText={setUsername}
        value={username}
      />

      <TextInput
        secureTextEntry={true}
        style={styles.input}
        placeholder="Password:"
        onChangeText={setPassword}
        value={password}
      />
      <MButton
        backgroundColor={config.indigo}
        text="Login"
        onPress={() => CheckLogin()}
      />
    </View>
  );
};

export default Login;
