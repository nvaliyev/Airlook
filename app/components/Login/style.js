import {StyleSheet} from 'react-native';
import { config } from '../../servises/utils';

export const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection:'column',
        backgroundColor: "#f7f9fc",
        justifyContent: 'center',
        alignContent:'center',
        alignItems:'center',
        
    },
    input:{
height:65,
backgroundColor:config.lilac,
alignSelf:'stretch',
paddingHorizontal:15,
marginHorizontal:25,
marginBottom:6,
borderRadius:5,
elevation:10,
shadowColor:'black',
borderColor:config.eminence,
    },

});

