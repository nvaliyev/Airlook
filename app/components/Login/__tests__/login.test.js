import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Login from '..';
describe(' Login', () => {
    test(' Login snapshot', () => {
        const renderer = new ShallowRenderer();
        const snapshot = renderer.render(< Login/>);
        expect(snapshot).toMatchSnapshot();
    });
});


