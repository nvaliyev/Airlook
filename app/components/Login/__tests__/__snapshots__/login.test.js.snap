// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Login  Login snapshot 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "backgroundColor": "#f7f9fc",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Username:"
    rejectResponderTermination={true}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#F7F9FC",
        "borderColor": "#6f347c",
        "borderRadius": 5,
        "elevation": 10,
        "height": 65,
        "marginBottom": 6,
        "marginHorizontal": 25,
        "paddingHorizontal": 15,
        "shadowColor": "black",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Password:"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#F7F9FC",
        "borderColor": "#6f347c",
        "borderRadius": 5,
        "elevation": 10,
        "height": 65,
        "marginBottom": 6,
        "marginHorizontal": 25,
        "paddingHorizontal": 15,
        "shadowColor": "black",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <ClickButton
    backgroundColor="#5d6fc4"
    onPress={[Function]}
    text="Login"
  />
</View>
`;
